src/octonode/client.coffee

```diff
@@ -142,37 +144,106 @@ class Client
         body = JSON.parse(body || '{}')
       catch err
         return callback(err)
-
+    debugger
     return callback(new HttpError(body.message, res.statusCode, res.headers, body)) if body.message and res.statusCode in [400, 401, 403, 404, 410, 422]
     callback null, res.statusCode, body, res.headers
 
+  errorHandlePromise: (response, cb) ->
+    debugger
+    {statusCode, body, headers} = response
+
+    fiveError = new HttpError('Error ' + statusCode, statusCode, headers)
+    fourError = new HttpError(body.message, statusCode, headers, body)
+
+    return cb(fiveError) if Math.floor(statusCode/100) is 5
+    return cb(fourError) if body.message and Math.floor(statusCode/100) is 4
```

Comment:
This was a leftover check from the original errorHandle function which first did a try/catch to convert body to JSON if the body was a string. Shouldn't be necessary any more because I'm forcing a JSON response using the request-promise options.